core/src/main/java/org/infinispan/atomic/impl/ApplyDelta.java

```diff
@@ -0,0 +1,93 @@
+package org.infinispan.atomic.impl;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+import org.infinispan.atomic.CopyableDeltaAware;
+import org.infinispan.atomic.Delta;
+import org.infinispan.atomic.DeltaAware;
+import org.infinispan.commons.CacheException;
+import org.infinispan.functional.EntryView;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.marshall.Ids;
+import org.infinispan.commons.marshall.Marshaller;
+import org.infinispan.commons.util.Util;
+import org.infinispan.factories.GlobalComponentRegistry;
+
+/**
+ * Replacement for {@link org.infinispan.commands.write.ApplyDeltaCommand} and
+ * {@link org.infinispan.context.Flag#DELTA_WRITE}. Deprecated since {@link Delta},
+ * {@link org.infinispan.atomic.DeltaAware} and {@link org.infinispan.atomic.CopyableDeltaAware} are deprecated, too.
```

Comment:
I don't think you really need the deprecation if it's not visible to applications.